### ISSUE TITLE
Extend the link_target insert tag to allow external URLs

### ISSUE THREAD

**Issue by SpeGal** [contao/contao#2802]
The insert tag {{link::*}} accepts external URLs however the insert tag {{link_target::*}} only accepts internal IDs (yes you could refer to a page within Contao that is set up as an external forwarding, but this would be an unpractical workarround).

Why not just do the same thing to "link_target" as you did to "link" back then [https://github.com/contao/core/issues/2842](url)

I have wondered about this soooo many times over the years. Finally I had time to make this request :)

**Comment by fritzmg** [comment]
`{{link::*}}` is replaced with

```
<a href="…">…</a>
```

and the `href` depends on the parameter. The parameter can be a page ID, a page alias or even an absolute URL.

`{{link_target::*}}` however is replaced with

```html
target="_blank" rel="noreferrer noopener"
```

but _only_ if the page given in the parameter is identified as an external page. Why would you want to use `{{link_target::https://example.org}}`? At this point you already know it's an external page and can use `target="_blank" rel="noreferrer noopener"` directly.

**Comment by SpeGal** [comment]
Let's take as an example the element type "table". The editors might not know HTML or if they know how to set up a link via HTML might not know either the targer="_blank" option and/or that they should use rel="noreferrer noopener".

Therefore an easy to use Insert Tag would come in handy. The editors have the list of Insert Tags already and know how to work with them. Just with external links they keep hitting the limit of the possible (just today I got a call from an editor again that wanted to do this very thing: external link in a table element (the tinymce editor is not available there) set to open a new tab and preferably being able to set their own visible text. 

It does not have to be the link_target-tag but logically it makes sence from an editors perspective.

It may as well be something like {{link_ext::https://www.xxx.xx::_blank}} or just {{link::https://www.xxx.xx::_blank}} or something alike.
Even better would be: {{link_blank::https://www.xxx.xx::Please click here for more information}}.

**Comment by fritzmg** [comment]
> The editors might not know HTML or if they know how to set up a link via HTML

But how would `{{link_target::https://example.org}}` help in that case? If you want to use that insert tag, you would still need to know HTML in order to use it:

```
<a href="https://example.org"{{link_target::https://example.org}}>https://example.org</a>
```





> It may as well be something like {{link_ext::[https://www.xxx.xx::_blank}}](https://www.xxx.xx::_blank%7D%7D) or just {{link::[https://www.xxx.xx::_blank}}](https://www.xxx.xx::_blank%7D%7D) or something alike.
> Even better would be: {{link_blank::https://www.xxx.xx::Please click here for more information}}.

Yeah, it sounds like you need a new type of insert tag altogether.

**Comment by m-vo** [comment]
You most likely want `noreferer` on external links and most certainly don't want `about: blank` anywhere. Now explain this to your editors. 😀

What I'm trying to say: IMHO you either need defaults that automagically act on all links or editors that know what they're doing. I'm not sure new insert tags would help either of these.

**Comment by SpeGal** [comment]
> But how would {{link_target::https://example.org}} help in that case? If you want to use that insert tag, you would still need to >know HTML in order to use it:

><a href="https://example.org"{{link_target::https://example.org}}>https://example.org</a>
 
The new version of the link_target-Tag would of course output a different code. {{link_target::https://example.org}} would become:

`<a href="https://example.org" target="_blank" rel="noreferrer noopener">Click here</a>`

The way the insert tag is programmed right now is kind of unneccassary anyways. Who creates a new page in the page tree to set it up as an external link just so you can use the link_target insert tag? Does really anybody use it this way? Why not just make a real-link-creatinginsert tag like suggested above? 

But I am also open for a new tag if the link_target-Tag was to be used widely (what I doubt). ;)

I can just say that years of experience with editors in mind, this feature would be a big help for them!

**Comment by fritzmg** [comment]
> The new version of the link_target-Tag would of course output a different code. {{link_target::[https://example.org}}](https://example.org%7D%7D) would become:
> 
> `<a href="https://example.org" target="_blank" rel="noreferrer noopener">Click here</a>`

That would be unintuitive. The point of the `{{link_target::*}}` insert tag is to _only_ output the ` target="…"` attribute (along with ` rel="…"` which was added later on).

**Comment by SpeGal** [comment]
I cannot see how it would be "unintuitive". The {{link}}-Tag does exactly that: create a full HTML link. I find it quite logical that {{link_target}} would do the same thing just with the target="_blank" added.

I think you understand my goal, right? Can you think of a different way to achieve it?

Make the {{link}}-Tag recognize that https:// means external link, hense add target="_blank" rel="noreferrer noopener" to the created link? Cause in almost every case you would want a new tab for an external link anyways.

**Comment by m-vo** [comment]
> Cause in almost every case you would want a new tab for an external link anyways.

It doesn't really matter what you (as editor) want. It's bad practice not letting the user decide what she wants.

**Comment by fritzmg** [comment]
> I think you understand my goal, right?

Of course, I am just saying that `{{link_target}}` is not the right insert tag for what you want in general. Its sole purpose is to output `target="_blank"` and nothing else. What you need is a completely new insert tag - or some additional parameters for existing ones may be.

**Comment by SpeGal** [comment]
> It's bad practice not letting the user decide what she wants.

Ohhh this argument. Who decides what is "bad practice"? It is a valid HTML5 attribute and has been standard for decades by now. Users are used to it and expect it. No bad practice about it. But I understand that you personally might have changed your preference (probably after reading a smart sounding web article). However that does not define or overwrite a well established standard. Goodness, even Goolgle News does it. They know what the users want (open all the articles in new tabs and then close them one by one once they are finished with them). For sooo many cases target _blank is a good option.  (no hard feelings) ;)

**Comment by SpeGal** [comment]
>  Its sole purpose is to output target="_blank" and nothing else.

That is exactly the reason why I think that nobody needs or uses this insert tag. And therefore it could find a new meaning and purpose in life :)   Or die and make room for an insert tag that people need ^^

**Comment by fritzmg** [comment]
> And therefore it could find a new meaning and purpose in life :)

The output of an insert tag cannot be changed. It needs to retain its backwards compatibility.

> That is exactly the reason why I think that nobody needs or uses this insert tag.

I don't know whether or not this insert tag is used by many users. I can only tell you its purpose: to build a custom link for a Contao page, e.g.:

```html
<a href="{{link_url::8}}" title="{{link_title::8}}" class="foo" data-foobar="foo"{{link_target::8}}>{{link_name::8}}</a>
```

**Comment by ausi** [comment]
> Goodness, even Goolgle News does it.

I cannot reproduce that 😊

**Comment by aschempp** [comment]
I didn't even know `{{link_target::*}}` existed until this ticket 😂 Maybe we could add something like a `{{link_external::*}}` for that purpose? It could also work with internal pages and make sure they always open in a new windows.

An alternative would be an insert tag flag, something like `{{link::*|external}}` (I like that one better actually).

**Comment by SpeGal** [comment]
> I didn't even know {{link_target::*}} existed until this ticket

Exactly, nobody needs it ^^


> It could also work with internal pages and make sure they always open in a new windows.

Maybe it should be called `{{link::*|blank}}` then. If it is for internal pages, too, 'external' might be confusing. 'Blank' might be easier to understand and remember.

Thank you for your constructive suggestion @aschempp! :)
